File: seesaw.py
# Print 1 if the seesaw is in balance, 0 otherwise.
# w1 * d1 == w2 * d2.

# 10kg      10kg
# |         |
# v         v
# ___________
#     A     

# Best answer Python
r = '10kg      10kg'.split()      # primera entrada, pasa de str a list
print(r);
for i in range(2):       # itera dos veces
    r[i] = int(r[i][:-2])# transforma a int, cada una elimina kg
a,b=r                    # asigna masas

exec('input();'*2)     # evade 2 inputs que son informacion no necesaria
le = len(input())      # Obtiene distancia total del balacin
f = len(input())       # obtiene distancia total del fulcrum
    
d1,d2 = f-1,le-f        # Asigna ditancias 
print((int(a*d1==d2*b)))# Pasa un booleano a int 

File: seesaw.js
// Print 1 if the seesaw is in balance, 0 otherwise.
// Distance is the distance between the mass and 
// the fulcrum('A')
//
// w1 * d1 == w2 * d2.
// 
// 10kg      10kg
// |         |
// v         v
// ___________
//     A      

let seesaw = ['10kg      10kg', '|         |', '___________', '    A'];
// console.log(seesaw[0].split(' '));
let [a, b] = seesaw[0].match(/\d+/g);
let total_distance = seesaw[2].length;
let ful = seesaw[3].indexOf('A')+1;
let d1 = ful-1;
let d2 = total_distance - ful;
console.log(d1*a == d2*b? 1: 0);